apps/studio/src/pages/sites/[siteId]/settings.tsx

```diff
@@ -1,8 +1,175 @@
+    }
+  }, [notificationFieldChanged, isOpen, onOpen, router.events])
+  return (
+    <>
+      <UnsavedSettingModal
+        isOpen={isOpen}
+        onClose={onClose}
+        nextURL={nextURL}
+      />
+      <Center pt="5.5rem">
+        <VStack w="48rem" alignItems="flex-start" spacing="1.5rem">
+          <Text w="full" textStyle="h3-semibold">
+            Manage site settings
+          </Text>
+          <Infobox textStyle="body-2" textColor="base.content.strong" size="sm">
+            Isomer Next is currently in Beta. To manage site settings that are
+            not displayed here, contact Isomer Support.
+          </Infobox>
+          <VStack w="full" alignItems="flex-start" mt="3rem" spacing="0.75rem">
+            <Text textStyle="h4">General</Text>
+            <HStack w="full" justifyContent="space-between" pt="0.5rem">
+              <Text textColor="base.content.strong" textStyle="subhead-1">
+                Display Site Notifications
+              </Text>
+              <Toggle
+                w="full"
+                label=""
+                isChecked={notificationEnabled}
+                onChange={(event) => {
+                  setNotificationFieldChanged(true)
+                  setNotificationEnabled(event.target.checked)
+                }}
+              />
+            </HStack>
+            <Text textColor="base.content.medium" textStyle="body-2">
+              The site notification will always be visible on the site until it
+              is dismissed by the user. <br />
+              Use a notification to inform users of key alerts. Do not use a
+              notification to advertise an event or a promotion.
+            </Text>
+            {notificationEnabled && (
+              <>
+                <Text
+                  textColor="base.content.strong"
+                  textStyle="subhead-1"
+                  pt="0.5rem"
+                >
+                  Notification Text
+                </Text>
+                <Input
+                  isDisabled={!notificationEnabled}
+                  placeholder="Notification should be succinct and clear"
+                  onChange={(event) => {
+                    setNotificationFieldChanged(true)
+                    setNotificationText(event.target.value)
+                  }}
+                  maxLength={100}
+                  value={notificationText}
+                />
+                <Text textColor="base.content.medium" textStyle="body-2">
+                  {remainingCharacters} characters left
+                </Text>
+              </>
+            )}
+          </VStack>
+          <HStack justifyContent="flex-end" w="full" gap="1.5rem">
+            {saveEnabled && (
+              <Text textColor="base.content.medium" textStyle="caption-2">
+                Changes will be reflected on your site immediately.
+              </Text>
+            )}
+            <Button onClick={onClickUpdate} isDisabled={!saveEnabled}>
```

Comment:
actually looking at this flow, it is actually a form right. any reason why react hook form is not used? That gives us validation handling out of the box, which the current implementation does not do yet.

Reply:
Hmm, no particular reason other than lack of awareness on my side on react hook form. Should I rework this to use react hook form?

Reply:
Please do!